### DISCUSSION TITLE
2023 Core Maintainer Elections

### DISCUSSION THREAD

**Discussion by TheCodeTraveler** [CommunityToolkit/Maui#1282] [General]
# Core Maintainer Elections 2023 

Greetings fellow Open Source developers! 

We are seeking new members to the .NET MAUI Community Toolkit Core Maintainer team, and we want YOU, the community, to decide who joins us. On the Community Toolkit team, we strongly believe that core maintainers should come from the community and be elected by the community. 

Going forward, we will hold yearly elections each year in July, finalizing the election results at the [2023 August Monthly Standup](https://github.com/CommunityToolkit/Maui/wiki/Monthly-Standups).

This year, 2023, we are seeking three (3) additional Core Maintainers.

### Nominations

All candidates are self-nominated.

Before nominating yourself, please read through our [Core Team Responsibilities](https://github.com/CommunityToolkit/Maui/wiki/Core-Team-Tools-and-Responsibilities#responsibilities) to ensure you are able to commit to the team for a minimum of one (1) year, are able to attend our monthly standup, and are able to contribute a minimum of one activity per week. 

To nominate yourself, write a top-level comment below in this Discussion introducing yourself to the community and highlighting why you'd make the best core maintainer!

Your self-nomination should include, but is not limited to, the following details about yourself:
- Years of open-source experience
- Years of .NET MAUI experience (can also include Xamarin experience)
- Links to contributions, bug reports, discussions (etc.) to the [CommunityToolkit organization](https://github.com/CommunityToolkit) 
  - Can include contributions to the [Xamarin.CommunityToolkit](https://github.com/xamarin/xamarincommunitytoolkit)
  - Can include contributions to the [CommunityToolkit.Maui.Markup](https://github.com/communitytoolkit/maui.markup)
- Links to other open-source projects you maintain
- New Proposals + Bug Fixes you will contribute to the .NET MAUI Community Toolkits
- Proficiency in English + Other languages 
  - The .NET MAUI Community Toolkit conducts all Issues, Pull Requests, Discussions, Standups, etc in English
  - A multi-lingual core maintainers is a HUGE bonus given the vast diversity of our community
    - Our current team covers Dutch, Portuguese, Spanish, Ukrainian and Korean

And similar to the [StackOverflow Moderator Elections](https://stackoverflow.com/election/14 ), we seek core core maintainers who exemplify the following traits:

- Patient and fair
- Leads by example
- Shows respect for their fellow community members in their actions and words
- Open to some light but firm moderation to keep the community on track, and resolve uncommon disputes and exceptions

The nominees who receive the most upvotes will join the .NET MAUI Community Toolkit Core Maintainer Team.

### Voting 

Voting is simple! [Upvote the nomination](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#upvoting-a-comment) for each Core Maintainer whom you wish to join the team. 

To vote for a candidate, read through the nominations, below, and [Upvote the nomination comment](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#upvoting-a-comment) for the candidate whom you'd like to join the team.

You are allows to vote (upvote) as many nominees as you wish.

At the end of the 2023 election, the nominations with the top three (3) total votes will join the team.

We ask the community to avoid abusing the voting system by creating new GitHub accounts to vote additional times for a preferred candidate. We reserve the right to ban users from the Community Toolkit organization whom we suspect of creating multiple accounts to vote for a specific candidate in the yearly elections.

### Results

Results will be finalized and posted here during the [2023 August Monthly Standup](https://github.com/CommunityToolkit/Maui/wiki/Monthly-Standups).

**Comment by mrlacey** [comment]
I might be missing something, but why use this approach rather than promote the most active/engaged contributors to "core" status?

If the goal is to bring in people who aren't yet contributing, what are the expectations, requirements, and responsibilities of core maintainers? Without knowing this, will people know what they are nominating themselves for?

**Reply by TheCodeTraveler** [reply]
Hey Matt! We've reached out privately to the biggest contributors. Most (all) don't have the bandwidth to join as a full-time maintainer. 

> If the goal is to bring in people who aren't yet contributing, what are the expectations, requirements, and responsibilities of core maintainers?

That is not the goal. Could you tell me what wording above implied that? I'll update the write-up accordingly. The goal is to allow the community to vote yearly for core maintainers. 

> Without knowing this, will people know what they are nominating themselves for?

We've included the [responsibilities for maintainers](https://github.com/CommunityToolkit/Maui/wiki/Core-Team-Tools-and-Responsibilities#responsibilities) in the top-level post

**Reply by mrlacey** [reply]
Having not seen an open-source project take this approach to defining/selecting/electing maintainers, I was trying to understand why this was being done. The lack of mention of reaching out to existing maintainers seemed notable in its omission (unless I missed it on the call.)
I was guessing at the motivation for the approach taken as I'm not even aware of any other projects of a similar size that have such a large group of existing core maintainers. To then assume that there will be a large enough number of people who aren't currently actively engaged but want to join (and are capable of joining) the core team that an election is needed is surprising. Especially with such a short time frame (results in one month.)
I don't want to come across as negative, more surprised and cynical. 
I'll be watching closely, but I can't nominate myself as I don't have the time to work on my own existing projects, let alone commit to something more.

**Comment by KSemenenko** [comment]
I have a bunch of free hands for open source (https://github.com/managedcode), and we can volunteer to contribute to this project that I love. However, generally speaking, at least for now, I don't think it's a good idea to invest time into MAUI. Perhaps this is why there aren't many maintainers.

Blazor works fine, but MAUI unfortunately does not. Maybe .NET 8 will be better. 

Anyway, here's my contribution to Xamarin: https://github.com/xamarin/Xamarin.Forms/issues/created_by/KSemenenko

Personally, I love MAUI and Xamarin, and I believe that one day they will be better than all their competitors.

**Reply by mrlacey** [reply]
It's great that you're providing time for people to work on this, but I don't see how it fits with your claim that you "don't think it's a good idea to invest time into MAUI."
Are you hoping that by contributing more to the community toolkit, it will get MAUI to a point that it is "worth investing time in"? Or is it something else?

**Reply by KSemenenko** [reply]
I think xamarin and now MAUI program model and platform is the best on market.
But a lot of work should be done before it will be 100% production ready.
And if I can help with it - I will.

**Comment by CliffAgius** [comment]
Hey team,

Been using Open Source and helping out on the Xamarin and MAUI repo's for a few years now and on the MAUI CT I have mostly been helping with Documentation and it's here that I would like to be considered for helping the team.

My commits to the Xamarin Repo started back in 2018 and for the Maui CT I was active for a bit as you can see with this example [PR](https://github.com/MicrosoftDocs/CommunityToolkit/pull/60) but due a course at the day job I had to step back for a little while.  Good news the course is over and I now have some time I can commit to the project.

However I do need to point out that my day job means that some months I may not make the stand-up as I could be off grid but I will try my upmost to be there each month.

As for new proposals I would really like to see the Camera view sorting I know this is an active issue at present but something I would like to help with as well.

As for me personally I am an Active community member and Microsoft MVP for Developer Technologies and can often be found at Meet-ups and Conferences talking about MAUI and Xamarin prior to that.  I work as a freelance dev mostly in the Mobile/MAUI and IoT areas and have 30+ years dev experience.

I know the current team but if your reading this and want to know more about me then please visit my (blog)[https://www.cliffordagius.co.uk/] or find me on (Twitter)[https://twitter.com/CliffordAgius]
